### DISCUSSION TITLE
using Xfreerdp3 to execute a RemoteApp is there a way to sequentially execute more than just 1 app

### DISCUSSION THREAD

**Discussion by bmullan** [FreeRDP/FreeRDP#10260] [Q&A]
In linux cli and in windows cli:

A & B will execute A first then when A completes it executes B
Is there a way to do that with xfreerdp's remoteapp syntax ?

**Comment by akallabeth** [comment]
just start one after the other?

the thing has tow drawbacks:
1. there is no way to get the exit status of a remote application
2. `xfreerdp` might need quite some time to disconnect after the application window closed, so that might not be ideal

**Comment by bmullan** [comment]
I see what you are saying... I'm a dolt

In my script I was actually executing xfreerdp first to run the remoteapp
then the next line executed xfreerdp again BUT this time executing Window's
C:\Windows\System32\logoff which I have also made a RemoteAPP so I can 
execute just that when needed.

That guarantees the complete previous session is terminated.

I was thinking that overcomes the problem of the hanging "fuse" issue not
letting the session totally close.

Thanks
Brian

**Reply by akallabeth** [reply]
what hanging fuse issue? the only one known was fixed pre 3.0?

**Reply by bmullan** [reply]
Have to backtrack... the "I'm a dolt" statement.   
I thought I knew what you were hinting to do but what I tried did not work.
However, that could still be me not using the command options correctily

Here is what I was doing originally & it works except on terminating the RemoteApp the command line just sits 
there until I CTRL-C

LOG_LEVEL=OFF \
/opt/freerdp-nightly/bin/xfreerdp3 \
/cert:ignore \
/gfx \
/rfx \
/h:640 \
/w:480 \
/u:$USER \
/bpp:32 \
/dynamic-resolution \
/printer \
/drive:auto,\* \
/drive:shared-folder,/home/bmullan/shared-folder \
/clipboard \
/sound:sys:pulse,format:1,quality:high,rate:44100,latency:50 \
**/app:program:'%windir%\explorer.exe'** \
/v:$win11ip     # gets set in the Bash script this is part of.

Here is what I tried based on what you said (note:  Previous to executing this one I had configured the "C:\Windows\System32\logoff
to also be a RemoteApp.

LOG_LEVEL=OFF \
/opt/freerdp-nightly/bin/xfreerdp3 \
/cert:ignore \
/gfx \
/rfx \
/h:640 \
/w:480 \
/u:$USER \
/bpp:32 \
/dynamic-resolution \
/printer \
/drive:auto,\* \
/drive:shared-folder,/home/bmullan/shared-folder \
/clipboard \
/sound:sys:pulse,format:1,quality:high,rate:44100,latency:50 \
**/app:program:'%windir%\explorer.exe & C:\Windows\System32\logoff'** \
/v:$win11ip

That did **not** work. 

But then I'm not sure using the **"&"** like that like  someone on the Windows Command line is correct but that is
what I thought you were trying to tell me when you said _**"just start one after the other?**"_

Then tried:

LOG_LEVEL=OFF \
/opt/freerdp-nightly/bin/xfreerdp3 \
/cert:ignore \
/gfx \
/rfx \
/h:640 \
/w:480 \
/u:$USER \
/bpp:32 \
/dynamic-resolution \
/printer \
/drive:auto,\* \
/drive:shared-folder,/home/bmullan/shared-folder \
/clipboard \
/sound:sys:pulse,format:1,quality:high,rate:44100,latency:50 \
**/app:program:'%windir%\explorer.exe'** \
**/app:program:'C:\Windows\System32\logoff'** \
/v:$win11ip

That **does not work** either which makes sense the **1st /app:program remoteapp (explorer)** first
has to fully terminate before the **2nd /app:program remoteapp** (C:\Windows\System32\logoff)
can run  & execute Window's "logoff" command.

But whatever is causing the **1st RemoteApp** execution to not fully terminate still requires a CTRL-C
which just terminates the Bash script

So do you have a cmd example of what "you" mean by **_Just start one after the other?_**

**Reply by akallabeth** [reply]
no, you can just start a program on the remote end (but that can be a script too ;))
there is currently no interface to start stuff on a running connection (you can, of course, just start the second command like the first, the first connection will disconnect and the second take over with both running)

NOTE:
* drop the `/bpp /gfx /rfx /h /w /dynamic-resolution /clipboard` stuff, that is counterproductive or already enabled by default
* drop the `:sys:pulse,format:1,quality:high,rate:44100,latency:50` that makes things worse in most cases, just let autodetection handle it

**Reply by bmullan** [reply]
@akallabeth 
I've used freerdp for years..
I had looked to find a link or something so I can donate to the project.  If it exists can you point me to it... thanks

**Comment by bmullan** [comment]
Thanks, I've not seen that information anywhere else before.   It will greatly simplify my xfreerdp3 command line

Also regarding your statement:

1. you can, of course, just start the second command like the first, the first connection will disconnect
2. and the second take over with both running

The problem I am seeing is related to your #1.   

I'd tried that but when the **_First RemoteApp_** starts and is later terminated 
_xfreerdp2/3 **does not** return to the command prompt without CTRL-C_

**So I implemented a useable Workaround.**

I made a copy of C:\Windows\System32\logoff so the User can click that and
completely terminate their Session.

**Reply by akallabeth** [reply]
then you should configure your server to log off sooner after rails termination.
`xfreerdp` terminates as soon as the server tells it the remote app session is closed.
